### ISSUE TITLE
Review "*Describer" singleton pattern for accessibility descriptions

### ISSUE THREAD

**Issue by zepumph** [phetsims/tasks#994]
From the singleton related design pattern meeting this week. 

A `*Describer.js` type is a type that deals with description related string formation/formatting, and concatenation. 

There is a good chance that the way that we are using describers should change. Up to this point we have been using them as singletons, but recent feedback suggests that instead they should be used just simple instances that can be passed into types that need it.

The current sims that use this pattern as singletons are:
- [x] friction
- [x] see https://github.com/phetsims/inverse-square-law-common/issues/63 ISLC suite
- [x] Molarity
- [x] Faraday's Law

It would be good to hammer out this pattern before it becomes any more prevalent. Tagging @jessegreenberg so he is aware of this issue, and tagging @pixelzoom for comments.

**Comment by pixelzoom** [comment]
After looking at SolutionDescriber in Molarity, I think the current Describer approach may be a misuse of the singleton pattern.  I agree with @samreid's recent observation that creating an instance (or instances) and passing them around is likely preferable. 

In the case of SolutionDescriber specifically, I don't see the advantage to putting everything into one describer.  The attempt to do so is responsible for some of the issues enumerated in https://github.com/phetsims/molarity/issues/67.   So I have yet to see convincing evidence that bundling things into a Describer is necessary or advantageous.

**Comment by zepumph** [comment]
@jessegreenberg and I discussed this today, and we think that both AlertManagers and Describers should be instances and not globals. In the general case, we want the flexibility to have a specific describer or alert manager to handle arbitrarily specific and complex types. For example an alert manager and describer specific to a Balloon in BASE, each instance of which would be tied to its own `BalloonModel`. I'll implement these changes in the above three sims, starting with Molarity later today with @twant. Thanks @jessegreenberg.

**Comment by pixelzoom** [comment]
For my general benefit...  Can you briefly describe "alert manager" and "describer", in terms of their responsibilities and scope?

**Comment by jessegreenberg** [comment]
"Describers" are responsible for generating the strings that describe the state of the model. In general, this will use `StringUtils.fillIn` extensively to create sentences that may change with model Properties. "Alert managers" are responsible for the dynamic alerts that happen separate from user focus, generally sending strings to `utteranceQueue`. "Alert managers" will often use "describers" to generate the alert strings.

**Comment by twant** [comment]
@zepumph and I did this for Molarity, and it went really smoothly. The only thing to work out is that the MolarityAlertManager isn't referenced anywhere else (an instance isn't needed), and it therefore creates a lint error. What should the pattern be, since sometimes we'll need to pass an AlertManager to other components, and sometimes we will not? We suppressed the lint rule manually for now, with a lint rule marked back to this issue. What do you think, @jessegreenberg?

**Comment by jessegreenberg** [comment]
@twant @zepumph what do you think about moving the work of MolarityAlertManager into MolarityScreenView and removing MolarityAlertManager? The only purpose of this alert manager is to organize/separate a11y code, and I wonder if that is indicative that this shouldn't be a class in this case.

**Comment by zepumph** [comment]
Sounds reasonable to me, though if there is a future increase in complexity/size in calls to utterance queue, then we can rethink it. @twant please take that on for molarity if that sounds good to you.

**Comment by twant** [comment]
@zepumph @jessegreenberg sounds good to me! I'll be working on Molarity tonight and tomorrow, and will implement then.

**Comment by zepumph** [comment]
FL actually didn't use this pattern, all of the describers are singletons. I fixed friction above. From recent pairings with TW, Molarity has been fixed. Closing.
